app/controllers/finance/financial_links_controller.rb

```diff
@@ -102,4 +116,17 @@ def find_financial_link
     @financial_link = FinancialLink.find(params[:id])
   end
 
+private
+
+  def financial_transaction_params
+    params.require(:financial_transaction).permit(:financial_transaction_type_id, :ordergroup_id, :amount, :note)
+  end
+
+  def find_best_fitting_ordergroup_id_for_financial_link(financial_link_id)
+    FinancialTransaction.joins(<<-SQL).order(created_on: :desc).pluck(:ordergroup_id).first
+      JOIN bank_transactions a ON financial_transactions.financial_link_id = a.financial_link_id
+      JOIN bank_transactions b ON a.iban = b.iban AND b.financial_link_id = #{financial_link_id}
```

Comment:
Thanks for separating this out!

We must be careful here to make sure this variable cannot be entered by the user.
In this case, it comes from `@financial_link`, which is a record loaded from the database, so it is fine.
Let's either add a note to the method about this, or add `.to_i`, which is tamper-proof in terms of SQL injection.
(I'm sorry I hadn't seen this earlier.)

Reply:
done, better safe than sorry